app/controllers/stats_controller.rb

```diff
@@ -57,11 +57,67 @@ def create_custom_report
     end
   end
 
-  def download_custom_report
-    report= Report.find(params[:id])
+  # RAG (Red, Amber, Green) thresholds for the different types
+  # Columns E, K and Q are percentages (non-trigger, trigger and overall)
+  # These are row items 4, 11 and 16.
+  RAG_THRESHOLDS_FOI = {red: 85, amber: 90}
+  RAG_THRESHOLDS_SAR = {red: 80, amber: 85}
+
+  COLOURED_ROWS = {'E' => 4, 'K' => 11, 'Q' => 16}
+
+  def cell_colour(value, thresholds)
+    if value < thresholds[:red]
+      'FF0000' #Red - bright
+    elsif value < thresholds[:amber]
+      'FFFF00' #Yellow - again very bright
+    else
+      '00FF00' #Green - quite bright (Lime)
+    end
+  end
+
+  def download_custom_report #rubocop:disable Metrics/MethodLength
+    report = Report.find(params[:id])
     filename = report.report_type.filename
 
-    send_data report.report_data, {filename: filename, disposition: :attachment}
+    # This is a Spike - so just hack the report about to show that an Excel report is possible
+    if report.report_type.abbr == 'R003'
+      # split report data into pieces - convert "" into empty string
+      excel_data = report.report_data.split("\n").map { |j| j.split(',') }
+      # Hack alert - SAR/FOI have different thresholds. This isn't really a good way of telling them apart
+      is_foi_report = excel_data[0][0].include?('FOI')
```

Comment:
I think SAR reports are r103 so we might be able use the report type to select the thresholds?